ItemsListApp/test/ItemsListApp.Api.UnitTests/Tests/Controllers/ItemsControllerUnitTests.cs

```diff
@@ -64,6 +63,35 @@ public void SetUp()
         }
 
         [Test]
+        public async Task Get_NonExistentId_ReturnsNull()
+        {
+            var itemId = new Guid("6341EB90-93E6-49AA-BBEC-2A69C3C8DB8D");
+            _itemsRepository.GetByIdAsync(itemId).Returns((Item) null);
+
+            var action = await _itemsController.GetAsync(itemId);
+            var response = await action.ExecuteAsync(CancellationToken.None);
+            Item actual;
+            response.TryGetContentValue(out actual);
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+            Assert.That(actual, Is.Null);
+        }
+
+        [Test]
+        public async Task Get_InvalidId_ReturnsBadRequest()
+        {
+            var invalidId = Guid.Empty;
+
+            var action = await _itemsController.GetAsync(invalidId);
+            var response = await action.ExecuteAsync(CancellationToken.None);
+            HttpError actual;
+            response.TryGetContentValue(out actual);
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(actual.Keys.Count, Is.GreaterThan(0));
+        }
+
+        [Test]
         public async Task Get_NoParameters_ReturnsAllItems()
```

Comment:
It is not necessary to add the following test, however, when testing a method that returns a collection of some items, it is no a bad habit to write an extra test method for no items being returned.

Methods returning `IEnumerable<>` should never return `null` by design, true, however, tests like this can prevent too optimistic calls (such as `_dependency.GetMeSomeNiceSurelyUnEmptyEnumerable().ToArray()[0].…`).